src/main/resources/log4j2.xml

```diff
@@ -28,10 +28,17 @@
                 <SizeBasedTriggeringPolicy size="20 MB"/>
             </Policies>
         </RollingRandomAccessFile>
+
+        <File name="HTMLAppender"
+              fileName="${DEFAULT_HOME}/userContent/Log-${date:MM-dd-k-m}.html"
+              append="false" >
+            <HTMLLayout charset="UTF-8" title="Audit Log" locationInfo="false" ></HTMLLayout>
+        </File>
     </Appenders>
     <Loggers>
-        <Logger name="AuditLogger" level="trace" additivity="false">
-            <AppenderRef ref="audit"/>
-        </Logger>
+        <Root level="trace" >
```

Comment:
What level are the audit events logged at? Trace seems somewhat low, and if Jenkins ever uses log4j-api to do its own logging, we'll end up including non-audit events in here. It would probably be a good idea to keep using the above `<Logger/>` tag.

Reply:
I understand the levels now from All, debug, info, warn, error, fatal, off, trace. Isn’t trace the right level?

Reply:
The problem here is that you're configuring the root logger to allow all log messages at trace or above to be logged (which are the rest of the logging levels). We are trying to ignore all other log messages besides audit events in this configuration so that any other log messages that inadvertently end up in our system that aren't audit events won't pollute our logs. This is more of a theoretical issue at the moment, but it would be an issue later on if log4j is used in more contexts in Jenkins.

You should keep the same `<Logger name="AuditLogger" level="trace" additivity="false">` instead of `<Root level="trace" >`.